server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -964,26 +1041,41 @@ public BucketCollectorProcessor bucketCollectorProcessor() {
     }
 
     /**
-     * Evaluate based on cluster and index settings if concurrent segment search should be used for this request context
-     * @return true: use concurrent search
-     *         false: otherwise
+     * Evaluate the concurrentSearchMode based on cluster and index settings if concurrent segment search
+     * should be used for this request context
+     * If the cluster.search.concurrent_segment_search.mode setting
+     * is not explicitly set, the evaluation falls back to the
+     * cluster.search.concurrent_segment_search.enabled boolean setting
+     * which will evaluate to true or false. This is then evaluated to "all" or "none" respectively
+     * @return one of "none", "auto", "all"
      */
-    private boolean evaluateConcurrentSegmentSearchSettings(Executor concurrentSearchExecutor) {
+    private String evaluateConcurrentSearchMode(Executor concurrentSearchExecutor) {
         // Do not use concurrent segment search for system indices or throttled requests. See:
         // https://github.com/opensearch-project/OpenSearch/issues/12951
         if (indexShard.isSystem() || indexShard.indexSettings().isSearchThrottled()) {
-            return false;
+            // return mode as "none"
+            return CONCURRENT_SEGMENT_SEARCH_MODE_NONE;
```

Comment:
```suggestion
            return CONCURRENT_SEGMENT_SEARCH_MODE_NONE;
```
Code speaks for itself